src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype130.cs

```diff
@@ -66,6 +82,64 @@ protected override void SaveProperties(Database db)
                     db.DatabaseScopedConfigurations[i].ValueForSecondary = this.currentState.databaseScopedConfigurations[i].ValueForSecondary;
                 }
             }
+
+            if (db.IsSupportedObject<QueryStoreOptions>())
+            {
+                try
+                {
+                    // Querystore propeties do not initialized at the time of create database
+                    var desiredState = this.currentState.queryStoreOptions.DesiredState;
+                }
+                catch (Exception ex)
+                {
+                    areQueryStorePropertiesInitialized = false;
```

Comment:
What exception are you expecting to be thrown here? I'm not sure what "initialized" would mean in this case, or why we'd want to skip setting the values completely in that case.

Reply:
While create database the value this.currentState.queryStoreOptions is not null but its properties (ex: DesiredState...etc) throws "'this.currentState.queryStoreOptions.DesiredState' threw an exception of type 'System.NullReferenceException'"